web/api/v1/api.go

```diff
@@ -889,6 +893,50 @@ func (api *API) serveFlags(r *http.Request) apiFuncResult {
 	return apiFuncResult{api.flagsMap, nil, nil, nil}
 }
 
+// Stat holdes the information about individual cardinality.
+type stat struct {
+	Name  string `json:"name"`
+	Value uint64 `json:"value"`
+}
+
+// TSDBStatus has information of cardinality statastics from postings.
+type tsdbStatus struct {
+	QueryDuration                    string `json:"queryTime"`
+	ScrapeValueCountByMetricsName    []stat `json:"scrapeValueCountByMetricsName"`
+	LabelValueCountByLabelName       []stat `json:"labelValueCountByLabelName"`
+	MemoryUsageInBytesByLabelName    []stat `json:"memoryUsageInBytesByLabelName"`
+	ScrapeValueCountByLabelValuePair []stat `json:"scrapeValueCountByLabelValuePair"`
+}
+
+func (api *API) serveTSDBStatus(r *http.Request) apiFuncResult {
+	db := api.db()
+	if db == nil {
+		return apiFuncResult{nil, &apiError{errorUnavailable, errors.New("TSDB not ready")}, nil, nil}
+	}
+	conver := func(stats []index.Stat) []stat {
+		result := make([]stat, 0, len(stats))
+		for _, item := range stats {
+			item := stat{Name: item.Name, Value: item.Count}
+			result = append(result, item)
+		}
+		return result
+	}
+
+	startTime := time.Now().UnixNano()
+	posting := db.Head().PostingsCardinalityStats("__name__")
+	totalQueryTime := fmt.Sprintf("%.3f", float64(time.Now().UnixNano()-startTime)/float64(1e9))
+
+	response := tsdbStatus{
+		QueryDuration:                    totalQueryTime,
+		ScrapeValueCountByMetricsName:    conver(posting.CardinalityMetricsStats),
```

Comment:
So we either can keep the original field names from the TSDB here, or at least the new ones need some adjustments. `Scrape...` is not quite correct, as data in the TSDB doesn't necessarily have to come from scrapes (it could theoretically come entirely from recording rules). If you do want to rename the field names, the following names would make most sense to me:

* `SeriesCountByMetricName`
* `LabelValueCountByLabelName` (as you have it)
* `MemoryInBytesByLabelName` (I think dropping `Usage` is ok to keep it shorter)
* `SeriesCountByLabelPair`